### ISSUE TITLE
Error with with jax v0.4.14

### ISSUE THREAD

**Issue by joverlee521** [blab/evofr#28]
evofr v0.1.20 raises the following error when using jax v0.4.14 (in the context of the run-mlr-model script in forecasts-ncov): 
```
[batch] [2023-07-31T18:14:08-07:00]   File "/nextstrain/build/./scripts/run-mlr-model.py", line 361, in <module>
[batch] [2023-07-31T18:14:08-07:00]     multi_posterior = fit_models(
[batch] [2023-07-31T18:14:08-07:00]   File "/nextstrain/build/./scripts/run-mlr-model.py", line 160, in fit_models
[batch] [2023-07-31T18:14:08-07:00]     posterior.save_posterior(f"{path}/models/hierarchical.json")
[batch] [2023-07-31T18:14:08-07:00]   File "/usr/local/lib/python3.10/site-packages/evofr/posterior/posterior_handler.py", line 37, in save_posterior
[batch] [2023-07-31T18:14:08-07:00]     json.dump(self.samples, file, cls=EvofrEncoder)
[batch] [2023-07-31T18:14:08-07:00]   File "/usr/local/lib/python3.10/json/__init__.py", line 179, in dump
[batch] [2023-07-31T18:14:08-07:00]     for chunk in iterable:
[batch] [2023-07-31T18:14:08-07:00]   File "/usr/local/lib/python3.10/json/encoder.py", line 431, in _iterencode
[batch] [2023-07-31T18:14:08-07:00]     yield from _iterencode_dict(o, _current_indent_level)
[batch] [2023-07-31T18:14:08-07:00]   File "/usr/local/lib/python3.10/json/encoder.py", line 405, in _iterencode_dict
[batch] [2023-07-31T18:14:08-07:00]     yield from chunks
[batch] [2023-07-31T18:14:08-07:00]   File "/usr/local/lib/python3.10/json/encoder.py", line 438, in _iterencode
[batch] [2023-07-31T18:14:08-07:00]     o = _default(o)
[batch] [2023-07-31T18:14:08-07:00]   File "/usr/local/lib/python3.10/site-packages/evofr/posterior/posterior_helpers.py", line 159, in default
[batch] [2023-07-31T18:14:08-07:00]     if isinstance(obj, jnp.DeviceArray):
[batch] [2023-07-31T18:14:08-07:00]   File "/usr/local/lib/python3.10/site-packages/jax/_src/deprecations.py", line 53, in getattr
[batch] [2023-07-31T18:14:08-07:00]     raise AttributeError(f"module {module!r} has no attribute {name!r}")
[batch] [2023-07-31T18:14:08-07:00] AttributeError: module 'jax.numpy' has no attribute 'DeviceArray'
```

According to the [CHANGELOG for v0.4.14](https://github.com/google/jax/blob/main/CHANGELOG.md#jax-0414-july-27-2023), `jax.numpy.DeviceArray` has been dropped and should be replaced with `jax.Array`. 

[`jax.Array` was introduced in v0.4.1](https://github.com/google/jax/blob/main/CHANGELOG.md#jax-041-dec-13-2022), which is the [minimum version required for evofr](https://github.com/blab/evofr/blob/24f8bdb17d593e2e9dc1c63ed61b0dbba236f749/pyproject.toml#L12), so it should be safe to just switch `jax.numpy.DeviceArray` to `jax.Array`.

**Comment by corneliusroemer** [comment]
I think this is resolved now? Can close this issue

**Comment by joverlee521** [comment]
Resolved by #29
